src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/cardgrid/viewmodel/CardGridViewModel.kt

```diff
@@ -0,0 +1,103 @@
+import com.github.thomasnield.rxkotlinfx.toObservable
+import io.reactivex.Observable
+import javafx.beans.property.SimpleBooleanProperty
+import javafx.collections.FXCollections
+import javafx.collections.ObservableList
+import org.wycliffeassociates.otter.common.data.model.ContentLabel
+import org.wycliffeassociates.otter.common.data.workbook.Chapter
+import org.wycliffeassociates.otter.common.data.workbook.Workbook
+import org.wycliffeassociates.otter.jvm.app.ui.cardgrid.CardData
+import org.wycliffeassociates.otter.jvm.app.ui.cardgrid.CardDataMapper
+import org.wycliffeassociates.otter.jvm.app.ui.workbook.viewmodel.WorkbookViewModel
+import org.wycliffeassociates.otter.jvm.utils.onChangeAndDoNow
+import tornadofx.*
+
+class CardGridViewModel: ViewModel() {
+    private val workbookViewModel: WorkbookViewModel by inject()
+
+    // List of content to display on the screen
+    // Boolean tracks whether the content has takes associated with it
+    val allContent: ObservableList<CardData>
+            = FXCollections.observableArrayList()
+    val filteredContent: ObservableList<CardData>
+            = FXCollections.observableArrayList()
+
+    // Whether the UI should show the plugin as active
+    private var showPluginActive: Boolean by property(false)
+    val showPluginActiveProperty = getProperty(CardGridViewModel::showPluginActive)
+
+    private var loading: Boolean by property(false)
+    val loadingProperty = getProperty(CardGridViewModel::loading)
+
+    val chapterModeEnabledProperty = SimpleBooleanProperty(false)
+
+    init {
+        Observable.merge(chapterModeEnabledProperty.toObservable(), allContent.changes()).subscribe { _ ->
+            filteredContent.setAll(
+                    if (chapterModeEnabledProperty.value == true) {
+                        allContent.filtered { it.item == ContentLabel.CHAPTER.value }
+                    } else {
+//                        allContent.filtered { it.item != ContentLabelEnum.CHAPTER.value }
+                        allContent
+                    }
+            )
+        }
+
+        workbookViewModel.activeWorkbookProperty.onChangeAndDoNow {
+            it?.let { wb -> loadChapters(wb) }
+        }
+
+        workbookViewModel.activeChapterProperty.onChange {
+            it?.let { ch -> loadChapterContents(ch) }
+                ?: workbookViewModel.activeWorkbookProperty.value
+                    ?.let { wb -> loadChapters(wb) }
+        }
+    }
+
+    private fun loadChapterContents(chapter: Chapter) {
+        // Remove existing content so the user knows they are outdated
+        allContent.clear()
+        loading = true
+        chapter.chunks
+            .map(CardDataMapper.Companion::mapChunkToCardData)
+            .doOnComplete {
+                loading = false
+            }
+            .observeOnFx()
+            .toList()
+            .doOnSuccess {
+                allContent.setAll(it)
+            }.subscribe()
```

Comment:
Replace doOnSuccess + subscribe with
```
            .subscribe { list: List<CardData> ->
                allContent.setAll(list)
            }
```
Something is funny about certain Single.subscribe overloads, so I needed to specify the type explicitly, and maybe this is why you used `doOnSuccess`.

Reply:
@KJoslyn what are your thoughts here?

Reply:
It was a while ago, but that may have been why I did it that way (there wouldn't be any other reason.) I just changed this and the other instance of this in the same file.